rstudio/packrat
# Update vendored `renv`; update script to allow taking tagged versions

This PR updates the embedded version of `renv` used by Packrat. Since the current `HEAD` on the `renv` repo is not passing CI, I took the previous version. To do this, I modified the `tools-sync-renv.R` script to allow passing a single argument, like so:

```bash
R -f tools/tools-sync-renv.R --args 0.15.5-48
```

If there are zero arguments, the script will proceed as before. If there is one argument, it uses that in a modified `git checkout` command. If there is more than one argument, it stops and emits a warning.

### QA notes

These are taken from the last "bump renv" PR.

We want to ensure that this change doesn't break dependency detection, which is currently the only place we use `renv`. One way to exercise this is to deploy a simple Shiny app to an RStudio Connect instance.

After installing the version of Packrat in this PR, deploy the following Shiny app to a Connect instance and ensure that it runs.

```R
library(shiny)

ui <- fluidPage(
    titlePanel("Old Faithful Geyser Data"),
    sidebarLayout(
        sidebarPanel(
            sliderInput("bins",
                        "Number of bins:",
                        min = 1,
                        max = 50,
                        value = 30),
            div("username", verbatimTextOutput("user")),
            div("groups", verbatimTextOutput("groups"))
        ),

        mainPanel(
            plotOutput("plot")
        )
    )
)

server <- function(input, output, session) {
  output$plot <- renderPlot({
    x    <- faithful[, 2]
    bins <- seq(min(x), max(x), length.out = input$bins + 1)

    hist(x, breaks = bins, col = 'darkgray', border = 'white')
  })

  output$user <- renderText(session$user)
  output$groups <- renderText(session$groups)
}

shinyApp(ui = ui, server = server)
```
